support/ebpf/types.h

```diff
@@ -571,8 +571,15 @@ typedef struct UnwindState {
 
   // Set if the PC is a return address. That is, it points to the next instruction
   // after a CALL instruction, and requires to be adjusted during symbolization.
-  // On aarch64, this additionally means that LR register can not be used.
   bool return_address;
+
+#if defined(__aarch64__)
+  // On aarch64, whether to allow LR-based unwinding.
+  // This is only allowed for leaf user-mode frames. Frames making a syscall
+  // are also considered leaf frames for this purpose, because LR is preserved across
+  // syscalls.
+  bool can_use_lr;
```

Comment:
nit: I would probably call this `lr_valid`. Or perhaps even invert it to `lr_invalid`  to make the potential helper be able to assign same value to `return_address` and `lr_invalid` in most code paths.